w3vm/vm.go

```diff
@@ -582,3 +582,40 @@ func WithFetcher(fetcher Fetcher) Option {
 func WithTB(tb testing.TB) Option {
 	return func(vm *VM) { vm.opts.tb = tb }
 }
+
+// Clone creates a new VM with a copy of the current VM's options and a snapshot of its state.
+func (vm *VM) Clone() (*VM, error) {
```

Comment:
I would prefer if the signature is simply
```go
func (vm *VM) Clone() *VM
```
and if instead of using options and `New` just copy the opts and create a `VM` instance directly. For many attributes like the fetcher a shallow copy is completely fine